news/12062-fix-name-overwrite

```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* Fixed a confusing case where [name=...] could overwrite the name specified outside of brackets. (#12062)
```

Comment:
How about something like "Prefer the outer name when a MatchSpec specifies a package's name twice `package-name[name=package-name]`"

Unfortunately we can write MatchSpec's without an outer package name?

Reply:
Yeah,  "*[name=...]" is an accepted MatchSpec.  Something of the flavor "*[sha256=...]" is the intended way to install the package with exactly that hash.